tap_invoiced/sync.py

```diff
@@ -54,6 +57,24 @@ def get_selected_streams(catalog):
             selected_streams.append(stream.tap_stream_id)
 
     return selected_streams
+@backoff.on_exception(
+    backoff.expo,
+    (HTTPError, ConnectionError, Timeout),
```

Comment:
Here we are handling generic exceptions but SDK handles those differently in it implementation.

Please refer [this implementation](https://github.com/Invoiced/invoiced-python/blob/master/invoiced/client.py#L145-#L174) and handle it accordingly.